src/WebApp/Pages/Account/Login.cshtml.cs

```diff
@@ -0,0 +1,68 @@
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+
+using IdentityServer4.Events;
+using IdentityServer4.Services;
+
+using Codidact.Authentication.Domain.Entities;
+using Codidact.Authentication.WebApp.Common;
+
+namespace Codidact.Authentication.WebApp.Pages.Account
+{
+    [SecurityHeaders]
+    [BindProperties]
+    public class LoginModel : PageModel
+    {
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly ILogger<LoginModel> _logger;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IEventService _events;
+
+        public LoginModel(SignInManager<ApplicationUser> signInManager,
+                          ILogger<LoginModel> logger,
+                          UserManager<ApplicationUser> userManager,
+                          IEventService events)
+        {
+            _signInManager = signInManager;
+            _logger = logger;
+            _userManager = userManager;
+            _events = events;
+        }
+
+        [Required, DataType(DataType.EmailAddress)]
+        public string Email { get; set; }
+
+        [Required, DataType(DataType.Password)]
+        public string Password { get; set; }
+
+        public bool RememberLogin { get; set; } = false;
+
+        public string ReturnUrl { get; set; } = "/index";
```

Comment:
Is this received from the request?

Reply:
I really thought that ASP.NET would figure this out somehow.

https://github.com/asynts/codidact-authentication/tree/1/move-from-core.wip